packages/delisp-core/src/compiler.ts

```diff
@@ -19,11 +19,19 @@ interface Environment {
   [symbol: string]: string;
 }
 
+// Convert a Delisp variable name to Javascript. This function should
+// be injective so there is no collisions and the output should be a
+// valid variable name.
+function varnameToJS(x: string): string {
+  // TODO: make it always a valid JS name!
+  return x;
```

Comment:
Would it make sense to, for example, prefix this with a unique number (global counter) for the time being? Feels a bit strange to propagate the shadowing behaviour of JS.

It makes the output a little bit more ugly (although adding the original variable name improves things already compared to the `p1` format), but that could be a point of improvement in the future.

Reply:
It is definitely possible, and it is what I have done before.

The problem I find is that it's very easy to go from shadowing to having global names.  But the opposite can be quite hard  (as you start assuming global names).

So I'd only go in that direction when we have a clear benefit of doing it.